packages/python/plotly/plotly/io/_base_renderers.py

```diff
@@ -669,6 +669,11 @@ def open_html_in_browser(html, using=None, new=0, autoraise=True):
     if isinstance(html, six.string_types):
         html = html.encode("utf8")
 
+    if isinstance(using, tuple):
+        using = [i for i in webbrowser._browsers.keys() if any(j in i for j in using)][
+            0
+        ]
```

Comment:
Is there a reason to use `if any(j in i for j in using)` here instead of `if i in using` here?  I think the way it is now, you would get partial string matches, which might be confusing.

Also, I think this will cause an index-out-of-bounds error if none of the browsers in the tuple are available.  I don't recall what error is currently raised, but it would be nice to report something a little more informative (e.g. A `ValueError` that explains that none of the elements in `using` were found in `list(if any(j in i for j in using))`).

Reply:
Yes @jonmmease using `if i in using` makes more sense than `if any(j in i for j in using)`, will get that change done. 

Regarding the second point I could do something like
```
if isinstance(using, tuple):
    try:
        using = [i for i in webbrowser._browsers.keys() if i in using][0]
    except TypeError:
         raise TypeError("""
Unable to detect any browsers in webbrowser._browsers.keys().
Check that the desired browser is installed on your system.
If problem still persists, try registering the browser using the python webbrowser module """)
```
Let me know if the TypeError is easy to understand or any suggestions are welcome

ValueError for invalid names like `chromee` is already handled and throws and error as follows `ValueError: 
Invalid named renderer(s) received: ['chromee']`

Reply:
I think sticking with raising a `ValueError` makes sense here (to match to existing behavior).  And I think you'll need to trap an `IndexError` rather than a `TypeError` to handle the case where no matches are found (this is the exception raised when you evaluate `[][0]`).

For the error message, I pictured including the actual list of supported browsers (the result of evaluating `list(webbrowser._browsers.keys())`) rather than "`webbrowser._browsers.keys()`") rather than itself.

Thanks!

Reply:
Hi @jonmmease can you please help me out with the content to put in the value error. Does this content seem ok? Any suggestions are welcome.
```
if isinstance(using, tuple):
    try:
        using = [i for i in webbrowser._browsers.keys() if i in using)][0]
    except IndexError:
        raise ValueError("""
Unable to detect the given browser.
Try one among the following "chrome", "chromium", "firefox" or "default" to use your default browser.
""")
```